cyral/model_integration_idp_generic_saml.go

```diff
@@ -0,0 +1,126 @@
+	Attributes               *RequiredUserAttributes `json:"requiredUserAttributes,omitempty"`
+	Completed                bool                    `json:"completed"`
+}
+
+type GenericSAMLSPMetadata struct {
+	XMLDocument string `json:"xmlDocument"`
+}
+
+type RequiredUserAttributes struct {
+	FirstName UserAttribute `json:"firstName,omitempty"`
+	LastName  UserAttribute `json:"lastName,omitempty"`
+	Email     UserAttribute `json:"email,omitempty"`
+	Groups    UserAttribute `json:"groups,omitempty"`
+}
+
+func NewRequiredUserAttributes(firstName, lastName, email, groups string) *RequiredUserAttributes {
+	return &RequiredUserAttributes{
+		FirstName: UserAttribute{
+			Name: firstName,
+		},
+		LastName: UserAttribute{
+			Name: lastName,
+		},
+		Email: UserAttribute{
+			Name: email,
+		},
+		Groups: UserAttribute{
+			Name: groups,
+		},
+	}
+}
+
+func (uatt *RequiredUserAttributes) WriteToSchema(d *schema.ResourceData) error {
+	var attributes []interface{}
+	attributes = append(attributes, map[string]interface{}{
+		"first_name": uatt.FirstName.Name,
+		"last_name":  uatt.LastName.Name,
+		"email":      uatt.Email.Name,
+		"groups":     uatt.Groups.Name,
+	})
+	return d.Set("attributes", attributes)
+}
+
+func RequiredUserAttributesFromSchema(d *schema.ResourceData) (*RequiredUserAttributes, error) {
+	attributesSet := d.Get("attributes").(*schema.Set).List()
+	if len(attributesSet) > 1 {
+		return nil, fmt.Errorf("Expected 'attributes' to be a set with at "+
+			"most one element, got %d elements", len(attributesSet))
+	} else if len(attributesSet) > 0 {
+		attributesMap := attributesSet[0].(map[string]interface{})
+		return NewRequiredUserAttributes(
+			attributesMap["first_name"].(string),
+			attributesMap["last_name"].(string),
+			attributesMap["email"].(string),
+			attributesMap["groups"].(string),
+		), nil
+	}
+	return nil, nil
+}
+
+type UserAttribute struct {
+	Name string `json:"name"`
+}
+
+type GenericSAMLIntegration struct {
+	ID            string                    `json:"id"`
+	DisplayName   string                    `json:"displayName"`
+	IdpType       string                    `json:"idpType"`
+	Disabled      bool                      `json:"disabled"`
+	IdpDescriptor *GenericSAMLIdpDescriptor `json:"idpDescriptor,omitempty"`
+	SPMetadata    *GenericSAMLSPMetadata    `json:"spMetadata,omitempty"`
+	Attributes    *RequiredUserAttributes   `json:"attributes"`
+}
+
+func (integ *GenericSAMLIntegration) WriteToSchema(d *schema.ResourceData) error {
+	d.SetId(integ.ID)
+	if integ.IdpDescriptor != nil {
+		if err := d.Set("single_sign_on_service_url",
+			integ.IdpDescriptor.SingleSignOnServiceURL,
+		); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (integ *GenericSAMLIntegration) ReadFromSchema(d *schema.ResourceData) error {
+	integ.ID = d.Id()
+	integ.IdpDescriptor = &GenericSAMLIdpDescriptor{
+		SingleSignOnServiceURL: d.Get("idp_metadata_url").(string),
```

Comment:
I believe this should be referencing the `single_sign_on_service_url` instead, right? Also, lets include the remaining field of the [SAMLIdentityProvider](https://github.com/cyralinc/integrations/blob/main/proto/generic-saml/generic_saml.proto#L367).
```suggestion
		SingleSignOnServiceURL: d.Get("single_sign_on_service_url").(string),
```

Reply:
I am including the other fields in the resource as part of the addition of the `idp_description` configuration option (see comment above). Will mark as resolved once that is done.